### PR TITLE
Frontend CLI tool

### LINKED ISSUE #13926: Frontend CLI tool
## Description
With the addition of e2e testing, the complexity of our package.json scripts and library configurations has increased due to the different needs of the environment it's under. We need a cli tool that handles everything needed for particular workflows to remove complexity in library configurations.

## Impact
- Saves FE engineers time in setting up the environment for E2E testing (required for starting the E2E UI tool, allows us to iterate a little faster and rerun them to ensure they are working correctly)
- Gives more flexibility in how we launch E2E testing

## Acceptance Criteria
- [ ] CLI tool handles the following workflows
  - [ ] dev
  - [ ] build
  - [ ] test
  - [ ] e2e
- [ ] Configuration in workflows depending on environment works as before

### PR DESCRIPTION
**Pull request by jpandersen87**
Fixes #13926

This PR adds a frontend CLI tool that helps facilitate workflows without relying on logic inside individual library configurations.

### CODE REVIEW

File: frontend-react/scripts/cli/cli.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    const env = loadRedirectedEnv(opts);
+    const childArgs = getChildArgs(process.argv);
+
+    // go straight to playwright if using help command
+    if (childArgs.includes("--help")) {
+        const child = spawn(
+            "playwright",
+            ["test", ...getChildArgs(process.argv)],
+            {
+                env,
```
Comment (penny-lischer) [inline frontend-react/scripts/cli/cli.ts:78]: Should we specify timezone here as well? or leave as part of playwright config

Comment (jpandersen87) [reply frontend-react/scripts/cli/cli.ts:81]: Good point. Playwright technically operates in two worlds at the same time (nodejs and the browser) so safer to make sure TZ is set here as well for the nodejs side.

**Review by penny-lischer — APPROVED** [review]
lgtm

File: frontend-react/scripts/cli/cli.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        "run local dev server (pass vite-specific commands after '--')",
+    );
+devCmd.action((_, cmd: Command) => {
+    process.env.NODE_ENV = "development";
+    const opts = cmd.optsWithGlobals();
+    const env = loadRedirectedEnv(opts);
+    spawnSync("vite", ["dev", ...getChildArgs(process.argv)], {
+        env,
+        stdio: "inherit",
+    });
```
Comment (etanb) [inline frontend-react/scripts/cli/cli.ts:34]: can we make sure it still spins up at port `3000`? right now it's spinning up at: `localhost:5173`

File: frontend-react/vite.config.ts [hunk 3]
```diff
@@ -12,64 +12,9 @@ import remarkFrontmatter from "remark-frontmatter";
...
-function getPort(mode: string) {
-    switch (mode) {
-        case "ci":
-        case "test":
-        case "csp":
-        case "preview":
-            return 4173;
-        default:
-            return 3000;
-    }
```
Comment (etanb) [inline frontend-react/vite.config.ts:46]: This is where that `3000` port is missing.

File: frontend-react/playwright.config.ts [hunk 4]
```diff
@@ -110,11 +103,4 @@ export default defineConfig<TestOptions>({
...
         // },
     ],
-    webServer: {
-        command: `yarn cross-env VITE_IDLE_TIMEOUT=25000 yarn run preview:build:${isCi ? "ci" : "test"}`,
-        url: "http://localhost:4173",
-        timeout: 1000 * 180,
-        stdout: "pipe",
-        // reuseExistingServer: !process.env.CI,
-    },
 });
```
Comment (etanb) [inline frontend-react/playwright.config.ts:106]: When I try running `npm run test:e2e-ui` or `CI=true npm run test:e2e-ui`, I keep getting the following error within Playwright:

```
Error: page.goto: net::ERR_ABORTED; maybe frame was detached?
Call log:
  - navigating to "http://localhost:4173/login", waiting until "domcontentloaded"
```

Comment (jpandersen87) [reply frontend-react/playwright.config.ts:106]: I'm assuming you meant `yarn run`? Try with the latest version and let me know if you still have issues.

**Review by etanb — APPROVED** [review]

**Pull request merged**